tiledb/sm/cpp_api/capi_string.h

```diff
@@ -49,17 +49,13 @@ namespace tiledb::impl {
 class CAPIString {
  public:
   CAPIString(tiledb_string_t** handle) {
-    if (handle == nullptr) {
+    if (handle == nullptr || *handle == nullptr) {
       throw std::invalid_argument("String handle cannot be null.");
     }
```

Comment:
There are two different ways for it to be invalid: a handle that's null and a non-null handle that contains null pointer. The exception message does not apply to the second, which represents an invalid handle (that's not a null handle).

Reply:
Reworded the exception message.